src/framework/theme/components/tree-grid/data-source/tree-grid-data.service.ts

```diff
@@ -3,22 +3,34 @@
  * Copyright Akveo. All Rights Reserved.
  * Licensed under the MIT License. See License.txt in the project root for license information.
  */
+
 import { Injectable } from '@angular/core';
 
-import { DEFAULT_ROW_LEVEL, NbTreeGridNode, NbTreeGridPresentationNode } from './tree-grid.model';
+import {
+  ChildrenGetter,
+  DataGetter,
+  DEFAULT_ROW_LEVEL,
+  ExpandedGetter,
+  NbTreeGridPresentationNode,
+} from './tree-grid.model';
 
 @Injectable()
 export class NbTreeGridDataService<T> {
 
-  toPresentationNodes(nodes: NbTreeGridNode<T>[], level: number = DEFAULT_ROW_LEVEL): NbTreeGridPresentationNode<T>[] {
-    return nodes.map((node: NbTreeGridNode<T>) => {
-      const presentationNode = new NbTreeGridPresentationNode(node, level);
-
-      if (node.children) {
-        presentationNode.children = this.toPresentationNodes(node.children, level + 1);
-      }
-
-      return presentationNode;
+  toPresentationNodes<N>(
+    nodes: N[],
+    dataGetter: DataGetter<N, T> = node => node.data,
```

Comment:
Also, let's move these default getters out of the function declaration. It's overcomplicated now, don't you think so?

Reply:
Done